paymentsheet-example/src/main/java/com/stripe/android/paymentsheet/example/viewmodel/PaymentSheetPlaygroundViewModel.kt

```diff
@@ -0,0 +1,95 @@
+package com.stripe.android.paymentsheet.example.viewmodel
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.switchMap
+import com.stripe.android.PaymentConfiguration
+import com.stripe.android.paymentsheet.PaymentSheet
+import com.stripe.android.paymentsheet.example.repository.DefaultRepository
+import com.stripe.android.paymentsheet.example.repository.Repository
+import com.stripe.android.paymentsheet.example.service.BackendApiFactory
+
+internal class PaymentSheetPlaygroundViewModel(
+    application: Application,
+    private val repository: Repository
+) : AndroidViewModel(application) {
+    val inProgress = MutableLiveData<Boolean>()
+    val status = MutableLiveData<String>()
+
+    val customerConfig = MutableLiveData<PaymentSheet.CustomerConfiguration?>()
+    val clientSecret = MutableLiveData<String?>()
+
+    val readyToCheckout: LiveData<Boolean> = customerConfig.switchMap { customerConfig ->
+        clientSecret.switchMap { clientSecret ->
+            MutableLiveData(customerConfig != null && clientSecret != null)
+        }
+    }
+
+    var checkoutMode: Repository.CheckoutMode? = null
+    var temporaryCustomerId: String? = null
+
+    suspend fun prepareCheckout(
```

Comment:
Feel like this method could use some docs, we should probably noted here that the checkout here is not standard Stripe mobile API and should usually done in the backend

Reply:
Also is it possible to refactor it with `BasePaymentSheetActivity#prepareCheckout` ?